AK/StringUtils.cpp

```diff
@@ -102,16 +103,19 @@ Optional<T> convert_to_int(StringView str, TrimWhitespace trim_whitespace)
             sign = -1;
     }
 
+    while (characters[i] == '0' && i < string.length())
+        ++i;
+
+    if ((string.length() - i) > NumericLimits<T>::max_digits())
+        return {};
```

Comment:
Just checking this does not mean we avoid overflow.
We can probably use a `max_digits_no_overflow` which is lower than your current `max_digits()`.

For example `9223372036854776808` has 19 digits but is higher than $2^{63}$.

This is a problem for every single integer type as powers of ten don't match the powers of two